File: function.py
#coding=utf-8

import time

from selenium import webdriver

commands = [
    {
        'command': 'browser',
        'parameters': {
            'url': 'http://ask.testfan.cn/',
        }
    },
    {
        'command': 'find',
        'parameters': {
            'by': 'css selector',
            'value':  'body > div.wrap > div.jumbotron.text-center > h4 > a.btn.btn-default.ml-5',
        }
    },
    {
        'command': 'action',
        'parameters': {
            'action': 'click',
            'sleep': 3
        }
    },
    {
        'command': 'find',
        'parameters': {
            'by': 'name',
            'value':  'email',
        }
    },
    {
        'command': 'keys',
        'parameters': {
            'value':  '3317434061@qq.com',
        }
    },
    {
        'command': 'find',
        'parameters': {
            'by': 'name',
            'value':  'password',
        }
    },
    {
        'command': 'keys',
        'parameters': {
            'value':  'damao@testfan',
        }
    },
    {
        'command': 'find',
        'parameters': {
            'by': 'xpath',
            'value':  '/html/body/div/div[2]/form/div[3]/button',
        }
    },
    {
        'command': 'action',
        'parameters': {
            'action': 'click',
            'sleep': 3
        }
    },
]

class Function(object):

    def __init__(self, commands):
        self.commands = commands
        self.driver = webdriver.Chrome(executable_path='C:\Python27\chromedriver.exe')

    def browser(self, params):
        url = params.get('url', "http://www.baidu.com")
        self.driver.get(url)

    def find(self, params):
        by = params.get('by')
        selector = params.get('value')
        return self.driver.find_element(by, selector)

    def keys(self, result, params):
        text = params.get('value')
        result.send_keys(text)

    def action(self, result, params):
        cmd = params.get('action')
        if cmd == 'click':
            result.click()

    def run(self):
        result = None
        for command in self.commands:
            cmd = command['command']
            params = command['parameters']
            if result is None:
                result = getattr(self, cmd)(params)
            else:
                result = getattr(self, cmd)(result, params)

if __name__ == '__main__':
    func = Function(commands)
    func.run()

File: static/js/function.js

function success(data) {
    console.log(data)
}

function fail(data) {
    console.log(data)
}

function browser() {
    var html = '\
    <div class="row" command="browser">\
        <label>打开：</label>\
        <input type="text" placeholder="http://www.testfan.com/">\
    </div>\
    '
    $('#cases').append(html)
}

function find() {
    var html = '\
    <div class="row" command="find">\
        <label>使用：</label>\
        <select>\
            <option value="xpath">xpath</option>\
            <option value="css selector">css selector</option>\
            <option value="name">name</option>\
        </select>\
        <label>参数：</label>\
        <input type="text">\
    </div>\
    '
    $('#cases').append(html)
}

function keys() {
    var html = '\
    <div class="row" command="keys">\
        <label>填写：</label>\
        <input type="text">\
    </div>\
    '
    $('#cases').append(html)
}

function action() {
    var html = '\
    <div class="row" command="action">\
        <label>动作：</label>\
        <select>\
            <option value="click">click</option>\
        </select>\
    </div>\
    '
    $('#cases').append(html)
}

function add_element() {
    var command = $('#option').val()
    if (command == 'browser') {
        browser();
    } else if (command == 'find') {
        find();
    } else if (command == 'keys') {
        keys()
    } else if (command == 'action') {
        action()
    } else {
        alert("错误的方法！")
    }
}

function parse_parameters(html) {
    var data = {}
    var command = $(html).attr("command")
    if (command == 'browser') {
        data['command'] = 'browser';
        data['parameters'] = {
            'url': $(html).find('input').val()
        }
    } else if (command == 'find') {
        data['command'] = 'find';
        data['parameters'] = {
            'by': $(html).find('select').val(),
            'value': $(html).find('input').val()
        }
    } else if (command == 'keys') {
        data['command'] = 'keys';
        data['parameters'] = {
            'value': $(html).find('input').val()
        }
    } else if (command == 'action') {
        data['command'] = 'action';
        data['parameters'] = {
            'action': $(html).find('select').val()
        }
    }else {
        console.log("错误的HTML！")
    }
    console.log(data)
    return data;
}

function run() {
    var list = $('#cases').find('div');
    var data = {
        'commands': []
    };
    $(list).each(function(index, item){
        var command = parse_parameters(item)
        data['commands'].push(command)
    });
    console.log(data)
    var url = 'http://127.0.0.1:9999/function/api';
    http(url, data, 'POST', success, fail);
}

$(function () {
    $('#command').click(add_element);
    $('#run').click(run);
});